Topic6/Exercise3/src/main/java/com/globant/repositories/UserRepository.java

```diff
@@ -0,0 +1,72 @@
+package com.globant.repositories;
+
+import java.util.ArrayList;
+
+import com.globant.models.User;
+
+/**
+ * This class allows to keep in-memory data of the users.
+ * 
+ * @author Yael Salim
+ *
+ */
+
+public class UserRepository {
+
+	ArrayList<User> userList;
+
+	public UserRepository() {
+
+		this.userList = new ArrayList<User>();
+	}
+
+	/**
+	 * Adds a new user to the database
+	 * 
+	 * @param newUser
+	 */
+	public void addUser(User newUser) {
+
+		this.userList.add(newUser);
+	}
+
+	/**
+	 * Deletes a user from the database
+	 * 
+	 * @param userName
+	 */
+	public void deleteUser(String userName) {
+
+		for (User actualUser : userList) {
+			if (userName.equals(actualUser.getName())) {
+				userList.remove(actualUser);
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Updates user's name
+	 * 
+	 * @param nickName
+	 * @param newName
+	 */
+	public void updateUser(String nickName, String newName) {
+
+		for (User actualUser : userList) {
+			if (nickName.equals(actualUser.getUsername())) {
+				userList.get(userList.indexOf(actualUser)).setName(newName);

```

Comment:
This is bad in complexity.. you already iterate over the collection to get the matched user, so is no needed to use the indexOf and get from list to updated the name of the element.